### PR TITLE
Sites: Don't bind selectors to state

### PR DESCRIPTION
**Pull request by mcsf**
Part of #14024 

This PR seeks to remove

```js
export default connect(
  ( state ) => ( {
    isSiteUpgradeable: isSiteUpgradeable.bind( null, state ),
  } )
)( Sites );
```

from `my-sites/sites`. @tyxla: You've been working a lot with Jetpack and capabilities, do you have recommendations on how to approach this? `Sites` passes a `filter` function prop to `SiteSelector`, which calls it to filter all site objects. How _wrong_ would it be to inspect the site objects instead of calling the selector?

### CODE REVIEW

**Comment by tyxla** [comment]
@mcsf I'm thinking we can add `isUpgradeable` as a computed prop. During the sites-list migration we've been trying to keep the computed props as less as possible, so I think we can add this one. We're already doing that for `is_previewable` anyway.

**Comment by mcsf** [comment]
@tyxla: sounds like a good idea. Would you be interested in picking it up, or should I?

**Comment by tyxla** [comment]
Sure, @mcsf, I've added a couple of commits to tackle that 👍

**Comment by mcsf** [comment]
Woot! Thanks!

**Comment by mcsf** [comment]
I can't formally approve this PR, since I created it, but LGTM!. :)

**Comment by tyxla** [comment]
Well, let's request some more 👀  here then 😉

**Comment by tyxla** [comment]
@gwwar good catch there!

In https://github.com/Automattic/wp-calypso/pull/14030/commits/1555eff39ecbe03813f2a27646b1ece1eb0a5fa2 I've applied a fix for that will allow to also include sites for which we can't determine whether they're upgradeable. Feel free to give the PR another try. Thanks!

File: client/my-sites/sites/sites.jsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const Sites = React.createClass( {
 
 		// Filter out sites with no upgrades on particular routes
 		if ( /^\/domains/.test( path ) || /^\/plans/.test( this.props.sourcePath ) ) {
-			return this.props.isSiteUpgradeable( site.ID );
+			return site.isSiteUpgradeable !== false;
```
Comment (gwwar) [inline client/my-sites/sites/sites.jsx]: This does the trick, but I wonder if we could be a bit more clear with:

```jsx
return ! site.isJetpack || site.isSiteUpgradable;
```

Comment (tyxla) [inline client/my-sites/sites/sites.jsx]: Sure, I'd go that way, but I was wondering: is there a case where upgradeable state can't be determined for a WordPress.com site? For example, if a site request fails to complete.

Comment (gwwar) [inline client/my-sites/sites/sites.jsx]: What does state look like in that case? I'd also be fine with the above and adding a clarifying comment so folks don't try and simplify the check.

Comment (tyxla) [inline client/my-sites/sites/sites.jsx]: After further consideration, I've decided to go with your suggestion here, it makes it much more readable than what I suggested.

**Comment by tyxla** [comment]
@mcsf that one is ready to go; do you want to deploy it, or should I? 😉

**Comment by mcsf** [comment]
@tyxla I feel you should do the honors, since I merely opened the PR and you did all the work :)

**Pull request merged**